src/datapoints-graph/charts/charts.component.ts

```diff
@@ -200,6 +202,38 @@ export class ChartsComponent implements OnChanges, OnInit, OnDestroy {
     }
   }
 
+  async exportAsCsv() {
+    this.chartOption$
+      .pipe(
+        first(),
+        map(({ series }) => {
+          if (!Array.isArray(series)) {
+            series = [series];
+          }
+
+          const csvData = series
+            .map(({ data, name, id }) => {
+              if (Array.isArray(data)) {
+                data.unshift([
+                  'Date',
+                  `"${name} ${
+                    id.toString().split('/').length === 2
+                      ? `(${id.toString().split('/')[1]})`
+                      : ''
+                  }"`,
+                ]);
+                return data;
+              }
+            })
+            .filter(Boolean);
+          return csvData;
```

Comment:
this whole mapping is csv related only, maybe it would be better to do in exportCsvService? to avoid subscribing in service, you can always make observable to promise.